client/dashboard/sites/overview/site-preview-card.tsx

```diff
@@ -0,0 +1,48 @@
+import { Card } from '@wordpress/components';
+import { useResizeObserver } from '@wordpress/compose';
+import SiteIcon from '../site-icon';
+import SitePreview from '../site-preview';
+import type { Site } from '../../data/types';
+
+const SitePreviewCard = ( { site }: { site: Site } ) => {
+	const [ resizeListener, { width, height } ] = useResizeObserver();
+	const { URL: url, is_a8c, is_private } = site;
+
+	// If the site is a private A8C site, X-Frame-Options is set to same origin.
+	const iframeDisabled = is_a8c && is_private;
+
+	return (
+		<Card>
+			{ resizeListener }
+			{ iframeDisabled && (
+				<div
+					style={ {
+						fontSize: '24px',
+						display: 'flex',
+						alignItems: 'center',
+						justifyContent: 'center',
+						height: '100%',
+					} }
+				>
+					<SiteIcon site={ site } />
+				</div>
+			) }
+			{ width && height && ! iframeDisabled && (
+				<div
+					style={ {
+						position: 'absolute',
+						top: 0,
+						right: 0,
+						bottom: 0,
+						left: 0,
+						overflow: 'hidden',
+					} }
+				>
+					<SitePreview url={ url } scale={ width / 1200 } height={ height / ( width / 1200 ) } />
```

Comment:
Is it in the scope of this PR to fix the site preview; it needs to have border-radius; currently it clashes with the card's border-radius

<img width="343" height="273" alt="Image" src="https://github.com/user-attachments/assets/d0f3327f-edf1-41ff-bacb-14d3d3c104a2" />